### PR TITLE
Tests: write a test to check for dupe locations

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Tests

## How was this tested?
I ran it against the commit where I was doing this :X

### CODE REVIEW

File: test/general/TestLocations.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from . import setup_default_world
+
+
+class TestBase(unittest.TestCase):
+    def testCreateDuplicateLocations(self):
+        for game_name, world_type in AutoWorldRegister.world_types.items():
+            if game_name in {"Final Fantasy"}:
+                continue
+            multiworld = setup_default_world(world_type)
+            locations = [location for location in multiworld.get_locations() if location.address is not None]
```
Comment (Berserker66) [inline test/general/TestLocations.py:12]: why are events excluded from this check?

File: test/general/TestLocations.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+            for location_name in world_type.location_name_to_id:
+                self.assertLessEqual(len([location for location in locations if location.name == location_name]), 1,
```
Comment (Berserker66) [inline test/general/TestLocations.py:14]: this is X*Y efficiency, you I suggest passing the names to collections.Counter() instead, then seeing if .most_common is > 1

File: test/general/TestLocations.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class TestBase(unittest.TestCase):
+    def testCreateDuplicateLocations(self):
+        for game_name, world_type in AutoWorldRegister.world_types.items():
+            if game_name in {"Final Fantasy"}:
+                continue
+            multiworld = setup_default_world(world_type)
+            locations = Counter([location for location in multiworld.get_locations()])
+            if len(locations) >= 1:
```
Comment (Berserker66) [inline test/general/TestLocations.py:14]:
```suggestion
            locations = Counter(multiworld.get_locations())
            if locations:
```
didn't test, but pretty sure this is all not needed.

Comment (alwaysintreble) [reply test/general/TestLocations.py:14]: about to check, almost did this, but i assumed that wouldn't work and just didn't lmao.

File: test/general/TestLocations.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+from . import setup_default_world
+
+
+class TestBase(unittest.TestCase):
+    def testCreateDuplicateLocations(self):
+        for game_name, world_type in AutoWorldRegister.world_types.items():
+            if game_name in {"Final Fantasy"}:
+                continue
+            multiworld = setup_default_world(world_type)
+            locations = Counter([location for location in multiworld.get_locations()])
```
Comment (Berserker66) [inline test/general/TestLocations.py:13]:
```suggestion
            locations = Counter(multiworld.get_locations())
```
Writing this suggestion again after it was invalidated by the last commit

Comment (alwaysintreble) [reply test/general/TestLocations.py]: 🤦

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**